src/components/menu/menu-trigger/menu-trigger.directive.ts

```diff
@@ -151,6 +158,31 @@ export class MenuTriggerDirective implements OnInit, OnDestroy {
         this._onDestroy$.complete();
     }
 
+    /** Focus the next focusable element */
+    focusNextElement() {
```

Comment:
I perhaps am not aware of the exact requirements regarding focus, but perhaps a more general `(close)`/`(closed)` event could be added in which case they could focus the desired element using it:

```html
<ux-filter-container closeOnBlur="true" (closed)="mySubmitButton.focus()"></ux-filter-container>
...
<button #mySubmitButton>Submit</button>
```

Just trying to avoid adding inputs for what seems like quite rare and specific use cases, whereas a `close` event is a bit more general and could be used for other things as well.

Reply:
Yeah agreed, I couldn't really think of any other way to achieve this. 

The current implementation works well as the closeAll event is fired then the tab event commences, meaning the trigger element gets focused, then the tab event fires and it moves focus to the next element. - https://github.com/UXAspects/UXAspects/blob/3b7260e78c5b9fc0895bfc0aafacdc68574786cf/src/components/menu/menu/menu.component.ts#L173-L175

The problem being this happens when the tab key is pressed and this is how we access the input in the dynamic filter. I tried experimenting with checking to see if the next element would be inside the menu but couldn't find a reliable solution.

I think the `focusElementOnClose` input is a good idea but I will mention the tabbing to next item was by request - https://teams.microsoft.com/l/message/19:f9c3e360e0454784a9fb04c49f04ea65@thread.skype/1642777403004?tenantId=856b813c-16e5-49a5-85ec-6f081e13b527&groupId=51fc2a42-4497-4231-baca-19348d1ef830&parentMessageId=1642777403004&teamName=%23ux-aspects&channelName=General&createdTime=1642777403004&allowXTenantAccess=false and I am not sure they would like to have to implement this for every menu they have.

Reply:
Fair enough - for a large number of instances it would be cumbersome to wire up elements like that. Just a thought anyway.

Reply:
I can experiment to check that the (closed) focus would work. Least that will offer the application team the ability to focus an element if the function doesn't work. I could also implement an input to specify the elements for the const on line 164?